src/bitwise/core.cljs

```diff
@@ -92,37 +57,49 @@
                            :text-decoration "none"
                            :color "black"})
 
-(defn process-info [architecture process]
-  (let [program (process->program process)
-        progress (r/atom 0)
-        runner-chan (process-runner architecture process)]
-    (go-loop [prog 0]
-      (reset! progress prog)
-      (recur (<! (:progress process))))
-    (fn []
-      (let [p @progress
-            pct (* p 100)]
-        [:div {:style (merge
-                       process-grid-styles
-                       {})}
-         [:div (:pid process)]
-         [:div (:name program)]
-         [:div {:style {:display "flex"}}
-          [:a {:on-click #(do (.preventDefault %) (async/offer! runner-chan :execute))
-               :role "button"
-               :href "#"
-               :style (merge
-                       action-button-styles
-                       {:background (str "linear-gradient(to right, lightgray " pct "%, white " pct "%)")})} "Execute"]
-          [:a {:on-click #(do (.preventDefault %)
-                              (dispatch! {:type :kill-process
-                                          :pid (:pid process)})
-                              (go (>! runner-chan :kill)))
-               :role "button"
-               :href "#"
-               :style (merge
-                       action-button-styles
-                       {:margin-left 10})} "Kill"]]]))))
+(defn action-button [props label]
+  [:a (util/merge-deep {:role "button"
+                        :href "#"
+                        :style action-button-styles}
+                       props
+                       {:on-click #(do (.preventDefault %) ((:on-click props)))})
+   label])
+
+(defn progress-button [props progress label]
+  (let [pct (* progress 100)]
+    [action-button
+     (util/merge-deep {:style {:background (str "linear-gradient(to right, lightgray " pct "%, white " pct "%)")}}
+                      props)
+     label]))
+
+(defn process-progress-button [props process label]
+  (let [progress (r/atom 0.0)]
+    (fn [props process label]
+      (reset! progress (process/progress process))
```

Comment:
This is the only way I could figure out of re-rendering the button while the job is in progress without leaking a go-loop or something when the component unmounts